tests/build/build_templates.go

```diff
@@ -55,7 +51,19 @@ func CreateComponent(ctrl *has.HasController, gitUrl, revision, applicationName,
 			},
 		},
 	}
-	c, err := ctrl.CreateComponent(componentObj, namespace, "", "", applicationName, false, constants.DefaultDockerBuildPipelineBundle)
+
+	var buildPipelineAnnotation map[string]string
+	if os.Getenv(constants.CUSTOM_SOURCE_BUILD_PIPELINE_BUNDLE_ENV) != "" {
+		customSourceBuildBundle := os.Getenv(constants.CUSTOM_SOURCE_BUILD_PIPELINE_BUNDLE_ENV)
+		Expect(customSourceBuildBundle).ShouldNot(BeEmpty())
+		buildPipelineAnnotation = map[string]string{
+			"build.appstudio.openshift.io/pipeline": fmt.Sprintf(`{"name":"java-builder", "bundle": "%s"}`, customSourceBuildBundle),
```

Comment:
Shouldn't it be `docker-build`, not `java-builder`?

(Although now that I think about it, when you specify the `bundle` the name doesn't really matter right?)

Reply:
In case of specific bundle, the name does not matter.
But to avoid confusion, I'll fix it.

Reply:
CreateComponent needs to handle use bundles for the list of components [here](https://github.com/konflux-ci/build-definitions/blob/3778abd242c09f1b19d67b46b1951d7c95849b9f/.tekton/tasks/e2e-test.yaml#L40) , also the docker-path can be different than default one for the components, will it be good to use some [scenarios file ](https://github.com/konflux-ci/e2e-tests/blob/main/tests/rhtap-demo/config/scenarios.go)like we use in rhtap-demo, deprecating [COMPONENT_REPO_URLS env](https://github.com/konflux-ci/build-definitions/blob/3778abd242c09f1b19d67b46b1951d7c95849b9f/.tekton/tasks/e2e-test.yaml#L40)